docs/technical/README.md

```diff
@@ -1,3 +1,45 @@
-# Technical Design Documentation
+# Technical Documentation
 
-Documents that explain technical design, architecture and interations of various Janssen modules
+This documentation explains technical design, architecture and interactions of various Janssen modules.
+
+Contents:
+
+- [API Reference](#api-reference)
+- [Design Consideration and Guidelines](#design-consideration-and-guidelines)
+  - [REST API Design](#rest-api-design)
+  - [Caching](#caching)
+  - [Testing](#testing)
+  - [Deployment](#deployment)
+  - [Security](#security)
+  - [Scalability and Cloud Infrastructure](#scalability-and-cloud-infrastructure)
```

Comment:
@yuriyz It would be great if you can contribute content for these guidelines. Since you have seen this design evolve over time, you would have the right background to drive this. Since these guidelines are applicable to Janssen in general, other code owners can also chip in too. @moabu @yurem

Reply:
Are these guidelines about AS or in general for any janssen project? 
E.g. AS is bound to spec and is developed according to it. While in jans-config-api we are free to build API which are not restricted by anything. 
If idea here is about general consideration then it should come from generic best practices which are not bound to any specific project (e.g. AS, cli, config-api). Otherwise each project needs own specific page where each code owner can describe current design and guidelines. Placing description of everything on single page looks overwhelming. What do you think?

Reply:
Got it. Yes, makes perfect sense. We can start with something and then during reviewes we can fill it with more and more info to not repeat it.

Reply:
:+1: 
Let's track this documetation effort in a separate issue.